core/arch/arm/kernel/generic_core_bootcfg.c

```diff
@@ -149,6 +146,9 @@ static struct map_area bootcfg_memory_map[] = {
 	{
 	 .type = DEVICE1_TYPE,
 	 .pa = DEVICE1_BASE, .size = DEVICE1_SIZE,
+#ifdef DEVICE1_VA_BASE
```

Comment:
@MrVan, what I'm saying is that we should always define both virtual and physical address in the map. It doesn't matter if they identical or not. If we do this then we can get rid of the special case where `va` isn't defined and we assign it from `pa`.